### DISCUSSION TITLE
Obvious Minion Usage Counter

### DISCUSSION THREAD

**Discussion by ptlthg** [EliteFarmers/Website#139] [Polls]
Would you support a system that approximates the ratio of a farmed crop to the amount generated from minions, rewarding less weight for minion generated crops?

This should be considered for a few reasons:
1. Given enough time, minion users (mostly melon is used) can obtain thousands of weight without farming
2. The total weight of a player is bloated by minions, not representative of time spent farming
3. Obvious profiles with almost no contest participations with millions of melon collection exist

The approximation formula for figuring out this ratio hasn't been worked on yet, but any final calculation would be underestimating the amount of minion generated drops, rather than overestimating. 

The way this would work is by looking at each past contest participations of a player, and work out what an estimated amount of time farming at 90% of that rate would come out to collection wise for `x` amount of hours before and after this contest. Then, the difference of this value versus maximum minion rates would be looked at, with the smallest of these values being considered.

We could then penalize this amount of crops gained by 50%, or some other value. **Again, the aim of this formula would be to underestimate minion drops, with small (relative) usage of minions possibly not even being detected.** For example, having 100 million crops gained from minions versus 1 billion crops manually farmed shouldn't have that 100 million penalized at all.

**Accuracy of results would be assessed before implementing this.**

_If you have another suggestion of how this could be implemented, please respond below!_

### Example, given Bankhier's current contests (very rough napkin math):

`25 Melon Contests` at roughly `550k Collection Average` is `~13.75 Million Melons Farmed (Guaranteed)`.
If we assume that Bankhier farmed a lot around each contest, say `3 hours of farming per contest (x9)` we come out to `123 Million Melons Farmed (Assumed)`. That is `123m / ~1.4b` total melon collection.

### Now let's do the same with Bankhier's Wheat collection:

`1,917 Wheat Contests` at roughly `200k Collection Average` is `~383 Million Wheat Farmed (Guaranteed)`. If we again assume that Bankhier farmed for 3 hours around each contest at those rates, we get `3.45 Billion Wheat Farmed (Assumed)` That is `3.45 Billion / 3.67 Billion (Actual Collection)`.

This is already looking fairly accurate, but using, say, 4 hours around each contest or higher could be used for much "safer" results. Unfortunately, due to how contests are random, this data is only the most accurate with a higher number of participations. Though, you have to also consider that a primarily farmed crop would have more entries to work with.

If we applied a penalty of 50% to the extra `~1.2 Billion` Melons that Bankhier collected, their melon weight would drop from around 4.2k to 2k. Only the resulting weight would change, and not the amount of collection displayed.

Share your thoughts!

**Comment by DarkenedFusion** [comment]
So I see where the issues are coming from, but I guess from my perspective, some crops that people tryhard in rely on minions. Weight is purely cosmetic, but crops like carrot have to be claimed pretty much every day  while giving approx. 10 or so FF compared to something like melon that after long enough waiting can give thousands. At the end of the day it doesn't matter too much, but just what I think from my pov.

**Reply by ptlthg** [reply]
I understand, what I proposed would likely not affect any collection gained from minions for any serious farmer of the crop in question. At least, I wouldn't be happy with a formula if this wasn't the case.

It's purely to find large discrepancies and punish those alone, which will likely end up only hurting people with melon minions and nothing else.

**Comment by spargelz** [comment]
As someone who only uses minions, I can not say anything else than keep it like it is.
